internal/resources/aws/cloudfront_distribution.go

```diff
@@ -0,0 +1,659 @@
+		ProductFilter: &schema.ProductFilter{
+			VendorName: strPtr("aws"),
+			Service:    strPtr("AmazonCloudFront"),
+			AttributeFilters: []*schema.AttributeFilter{
+				{Key: "requestDescription", Value: strPtr("Origin Shield Requests")},
+				{Key: "location", Value: strPtr(apiRegion)},
+			},
+		},
+	})
+
+	return costComponents
+}
+
+func (r *CloudfrontDistribution) invalidationRequestsCostComponents() []*schema.CostComponent {
+	costComponents := []*schema.CostComponent{}
+
+	var freeQuantity *decimal.Decimal
+	var paidQuantity *decimal.Decimal
+	if r.MonthlyInvalidationRequests != nil {
+		usageAmount := *r.MonthlyInvalidationRequests
+		if usageAmount < 1000 {
+			freeQuantity = decimalPtr(decimal.NewFromInt(usageAmount))
+		} else {
+			freeQuantity = decimalPtr(decimal.NewFromInt(1000))
+			paidQuantity = decimalPtr(decimal.NewFromInt(usageAmount - 1000))
+		}
+	}
+
+	costComponents = append(costComponents, &schema.CostComponent{
+		Name:            "Invalidation requests (first 1k)",
+		Unit:            "paths",
+		UnitMultiplier:  decimal.NewFromInt(1),
+		MonthlyQuantity: freeQuantity,
+		ProductFilter: &schema.ProductFilter{
+			VendorName: strPtr("aws"),
+			Service:    strPtr("AmazonCloudFront"),
+			AttributeFilters: []*schema.AttributeFilter{
+				{Key: "usagetype", Value: strPtr("Invalidations")},
+			},
+		},
+		PriceFilter: &schema.PriceFilter{
+			StartUsageAmount: strPtr("0"),
+		},
+	})
+
+	if paidQuantity != nil {
+		costComponents = append(costComponents, &schema.CostComponent{
+			Name:            "Invalidation requests (over 1k)",
+			Unit:            "paths",
+			UnitMultiplier:  decimal.NewFromInt(1),
+			MonthlyQuantity: paidQuantity,
+			ProductFilter: &schema.ProductFilter{
+				VendorName: strPtr("aws"),
+				Service:    strPtr("AmazonCloudFront"),
+				AttributeFilters: []*schema.AttributeFilter{
+					{Key: "usagetype", Value: strPtr("Invalidations")},
+				},
+			},
+			PriceFilter: &schema.PriceFilter{
+				StartUsageAmount: strPtr("1000"),
+			},
+		})
+	}
+
+	return costComponents
+}
+
+func (r *CloudfrontDistribution) encryptionRequestsCostComponents() []*schema.CostComponent {
+	costComponents := []*schema.CostComponent{}
+
+	if !r.HasFieldLevelEncryptionID {
+		return costComponents
+	}
+
+	var quantity *decimal.Decimal
+	if r.MonthlyEncryptionRequests != nil {
+		quantity = decimalPtr(decimal.NewFromInt(*r.MonthlyEncryptionRequests))
+	}
+
+	costComponents = append(costComponents, &schema.CostComponent{
+		Name:            "Field level encryption requests",
+		Unit:            "10k requests",
+		UnitMultiplier:  decimal.NewFromInt(10000),
+		MonthlyQuantity: quantity,
+		ProductFilter: &schema.ProductFilter{
+			VendorName: strPtr("aws"),
+			Service:    strPtr("AmazonCloudFront"),
+			AttributeFilters: []*schema.AttributeFilter{
+				{Key: "requestDescription", Value: strPtr("HTTPS Proxy requests with Field Level Encryption")},
+				{Key: "location", Value: strPtr("Europe")},
```

Comment:
The "Europe" location has been used in original code. The pricing appears to be the same for all regions.